### PR TITLE
Fix #120: Add check for actual authentication method availability

### LINKED ISSUE #120: Add check for actual authentication method availability
Currently, we only check user preferences to determine availability of an authentication method. For example, if user activates mobile token, we show a mobile token UI despite the fact user has no active devices.

We should add a generic `isAvailable` check to authentication method base, that would use user preferences as one of the inputs, but also check the actual method availability based on user and operation data...

```java
boolean isAvailable(String userId, String operationId);
```

### PR DESCRIPTION
**Pull request by romanstrobl**
This pull request fixes issues #120 and #98 related to dynamic change of availability of authentication methods e.g. when activations get blocked.

I also added error handling for initialization of authentication methods and fixed minor issues such as javadoc and i18n messages.

### CODE REVIEW

**Comment by romanstrobl** [comment]
Please wait with the review, I want to clean up the error handling.

**Comment by romanstrobl** [comment]
The pull request is ready for review now.

File: powerauth-webflow-authentication-operation-review/src/main/java/io/getlime/security/powerauth/lib/webflow/authentication/method/operation/controller/OperationReviewController.java [hunk 1]
```diff
@@ -61,14 +60,21 @@
...
     private final String FIELD_CHOSEN_AUTH_METHOD = "chosenAuthMethod";
 
-    @Autowired
-    private DataAdapterClient dataAdapterClient;
+    private final DataAdapterClient dataAdapterClient;
+    private final NextStepClient nextStepClient;
 
-    @Autowired
-    private NextStepClient nextStepClient;
+    public OperationReviewController(DataAdapterClient dataAdapterClient, NextStepClient nextStepClient) {
```
Comment (petrdvorak) [inline powerauth-webflow-authentication-operation-review/src/main/java/io/getlime/security/powerauth/lib/webflow/authentication/method/operation/controller/OperationReviewController.java]: Missing `@Autowired` above the constructor. If it works, it is by an accident.

**Comment by romanstrobl** [comment]
I added the missing @Autowired annotation.

**Pull request merged**